### PR TITLE
Pilot 1457: add new client for lineage

### PR DESCRIPTION
**Pull request by colorzzr**
## Summary

Add new api to support lineage operation:
 - add new entity
 - add lineage between entity
 - get lineage

## JIRA Issues

PILOT-1457
PILOT-1493

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

## Test Directions

(Additional instructions for how to run tests or validate functionality if not covered by unit tests)

### CODE REVIEW

File: common/lineage/entity_object.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import copy
+import time
+
+
+class FileDataAttribute:
```
Comment (Samantha77) [inline common/lineage/entity_object.py:20]: @colorzzr Could we redefine the data model for file entity? Some properties are deprecated already

File: common/lineage/lineage_client.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+class LineageClient:
+    def __init__(self, atlas_endpoint: str, username: str, password: str) -> None:
+        self.atlas_endpoint = atlas_endpoint
+        self.username = username
+        self.password = password
+
+        self.headers = {'content-type': 'application/json'}
+
+        self.logger = LoggerFactory('api_file_data').get_logger()
```
Comment (Samantha77) [inline common/lineage/lineage_client.py:37]: Could we rename the logger? "api_file_data" doesn't related to lineage operation

File: common/lineage/lineage_client.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        self.logger.setLevel(ERROR)
```
Comment (Samantha77) [inline common/lineage/lineage_client.py:38]: Logger level should be configurable

File: common/lineage/lineage_client.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                timeout=60,
+            )
+            if response.status_code != 200:
+                error_msg = f'Fail to create entity in Atlas with error: {response.text}'
+                self.logger.error(error_msg)
+                raise Exception(error_msg)
+
+        return response.json()
+
+    async def delete(self):
```
Comment (Samantha77) [inline common/lineage/lineage_client.py:113]: The delete operation should be able to logically delete an entity in Atlas

File: common/lineage/entity_object.py (hunk 1 above)
Comment (colorzzr) [reply common/lineage/entity_object.py:20]: yes I am going to propose one on the architecture review.

File: common/lineage/lineage_client.py (hunk 4 above)
Comment (colorzzr) [reply common/lineage/lineage_client.py:113]: for delete, I am not sure if we will do the logically delete or just change `archive` to `True`

File: common/lineage/lineage_client.py (hunk 3 above)
Comment (colorzzr) [reply common/lineage/lineage_client.py:38]: yes the client can use `debug_on` or `debug_off` to set the log level. `ERROR` is the default setting

File: common/lineage/lineage_client.py (hunk 4 above)
Comment (Samantha77) [reply common/lineage/lineage_client.py:113]: Will create seperate ticket to track down the delete function implementation.

**Review by Samantha77 — APPROVED** [review]

**Review by Samantha77 — APPROVED** [review]

**Review by Samantha77 — APPROVED** [review]
LGTM

**Pull request merged**